modules/imgproc/src/filter.simd.hpp

```diff
@@ -303,8 +305,9 @@ void FilterEngine__apply(FilterEngine& this_, const Mat& src, Mat& dst, const Si
 
     FilterEngine__start(this_, wsz, src.size(), ofs);
     int y = this_.startY - ofs.y;
+    ptrdiff_t offset = y * (ptrdiff_t)src.step;
     FilterEngine__proceed(this_,
-            src.ptr() + y*src.step,
+            src.ptr() + offset,
```

Comment:
we have tons of such code in OpenCV; are we going to refactor all of it?

Reply:
Yes, we should. It's a bug. Pointer overflow is disallowed by C++.

I only use a few OpenCV classes, but I have not encountered any other such overflows in my own use.  Are you sure there are so many really?

Reply:
Just `src.ptr() + y * (ptrdiff_t)src.step` should be enough.

Reply:
Done